### ISSUE TITLE
[FEATURE REQ] ServiceBusProcessor allow to adjust the prefetch count forUpdateConcurrency

### ISSUE THREAD

**Issue by danielmarbach** [Azure/azure-sdk-for-net#31624]
### Library name

Azure.Messaging.ServiceBus

### Please describe the feature.

The ServiceBusProcessor allows to dynamically adjust the concurrency by calling `UpdateConcurrency` which is a pretty nice feature!

Sometimes though when you are for example downsizing the concurrency it also makes sense to adjust the prefetch count to avoid running into too many MessageLockLostExceptions during the time window when running with limited concurrency. 

Since the prefetch count is a link setting, it would require to introduce an `ValueTask UpdateConcurrencyAsync(int maxConcurrentCalls, int? prefetchCount = default)` that when the prefetch count is set to a new value stops the behind-the-scenes receiver and re-establishes a new link with the new prefetch count. 

Currently, we are solving this scenario by stopping the receiver, creating a new one and starting the new one. Having this possibility on the processor would be :nail_care:

**Comment by jsquire** [comment]
Hi @danielmarbach.   Interesting idea, and I think that it makes sense conceptually.  If you're reading messages more slowly there's a bigger risk of having locks expire while sitting in prefetch.   @JoshLove-msft - thoughts?

**Comment by JoshLove-msft** [comment]
I think it makes sense. We have a similar request that is asking for the ability to modify this in the functions extensions - https://github.com/Azure/azure-sdk-for-net/issues/25929.

**Comment by danielmarbach** [comment]
By the way, I'd be happy to give this ago after my vacation sometime in late October, in case you find this valuable.

**Comment by JoshLove-msft** [comment]
> By the way, I'd be happy to give this ago after my vacation sometime in late October, in case you find this valuable.

Yes, that would be great! One thing we may want to think about is whether we should expose this for the regular receive and function extensions as well.

**Comment by danielmarbach** [comment]
I have raised last Friday https://github.com/Azure/azure-sdk-for-net/pull/32572 to get the discussion started.
